pkg/metainfo/kvmetainfo/stream.go

```diff
@@ -62,17 +64,40 @@ func (stream *readonlyStream) segment(ctx context.Context, index int64) (info st
 		copy(segment.EncryptedKeyNonce[:], segmentMeta.KeyNonce)
 		segment.EncryptedKey = segmentMeta.EncryptedKey
 	} else {
+		segmentPath = storj.JoinPaths("l", stream.encryptedPath)
 		segment.Size = stream.info.LastSegment.Size
 		segment.EncryptedKeyNonce = stream.info.LastSegment.EncryptedKeyNonce
 		segment.EncryptedKey = stream.info.LastSegment.EncryptedKey
 	}
 
-	var nonce storj.Nonce
-	_, err = encryption.Increment(&nonce, index+1)
+	contentKey, err := encryption.DecryptKey(segment.EncryptedKey, stream.Info().EncryptionScheme.Cipher, stream.streamKey, &segment.EncryptedKeyNonce)
```

Comment:
Instead of `stream.Info()` use `stream.info` directly. Less optimization work for compiler.